legion/engine/physics/physics_statics.cpp

```diff
@@ -341,5 +287,728 @@ namespace legion::physics
+
+            HalfEdgeFace* face = new HalfEdgeFace(pairing, faceNormal);
+
+            //push new face into list 
+            createdFaces.push_back(face);
+
+            //connect to 
+            if (pairingToConnectTo)
+            {
+                prevPairing->setPairingEdge(pairingToConnectTo);
+            }
+            else
+            {
+                initialPairing = prevPairing;
+            }
+
+            pairingToConnectTo = nextPairing;
+        }
+
+        initialPairing->setPairingEdge(pairingToConnectTo);
+    }
+
+    bool PhysicsStatics::foundFaceWithOutsideVert(std::list<ColliderFaceToVert>& facesWithOutsideVerts, PointerEncapsulator< ColliderFaceToVert>& outChosenFace)
+    {
+        for (auto& faceWithOutsideVert : facesWithOutsideVerts)
+        {
+            if (!faceWithOutsideVert.outsideVerts.empty())
+            {
+                outChosenFace.ptr = &faceWithOutsideVert;
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    void PhysicsStatics::partitionVerticesToList(const std::vector<math::vec3> vertices,
+        std::list<ColliderFaceToVert>& outFacesWithOutsideVerts)
+    {
+        for (const math::vec3& vertex : vertices)
+        {
+            ColliderFaceToVert* bestFaceToVert = nullptr;
+            float bestMatchDistance = std::numeric_limits<float>::lowest();
+
+            //find the face that is most distant from the std::list of planes given by 'outFacesWithOutsideVerts'
+            for (ColliderFaceToVert& faceToVert : outFacesWithOutsideVerts)
+            {
+                auto face = faceToVert.face;
+                float currentDistance = PointDistanceToPlane(face->normal, face->centroid, vertex);
+
+                if (currentDistance > bestMatchDistance)
+                {
+                    bestMatchDistance = currentDistance;
+                    bestFaceToVert = &faceToVert;
+                }
+            }
+
+            if (bestFaceToVert && bestMatchDistance > 0.0f)
+            {
+                bestFaceToVert->outsideVerts.push_back(vertex);
+            }
+        }
+    }
+
+    void PhysicsStatics::findHorizonEdgesFromFaces(const math::vec3& eyePoint, std::vector<HalfEdgeFace*>& faces,
+        std::vector<HalfEdgeEdge*>& outHorizonEdges, float scalingEpsilon)
+    {
+        //[1] Find first horizon edge
+        HalfEdgeEdge* initialHorizon = nullptr;
+        
+        auto findFirstHorizon = [&eyePoint,&initialHorizon,scalingEpsilon](HalfEdgeEdge*edge)
+        {
+            if (initialHorizon) { return; }
+           
+            if (edge->isEdgeHorizonFromVertex(eyePoint, scalingEpsilon))
+            {
+                initialHorizon = edge;
+            }
+        };
+
+        for (auto face : faces)
+        {
+            if (initialHorizon) { break; }
+
+            face->forEachEdge(findFirstHorizon);
+            
+        }
+
+        assert(initialHorizon);
```

Comment:
at some point, maybe with the refactor, this should probably get properly handled using `common::result` instead of crashing the entire engine

Reply:
would you prefer me to handle this in this pr or in an another separate pr?